internal/guest/runtime/hcsv2/uvm.go

```diff
@@ -318,7 +331,7 @@ func (h *Host) CreateContainer(ctx context.Context, id string, settings *prot.VM
 		}
 	}
 
-	h.containers[id] = c
+	c.status = containerCreated
```

Comment:
I think we need to do this under lock as well, otherwise we could be changing `c.status` just as another thread is reading it.

Reply:
container level lock I assume?

Reply:
It could be the map lock, as long as we ensure only one accessor. You could also look at using an atomic here.

Reply:
currently that isn't possible with the existing implementation but it was a concern that i raised with @anmaxvl as something to look at for the future if not wasn't addressed now.

Reply:
changed to atomic update. had to switch to uint32 without custom type definition.